packages/snaps-simulation/src/structs.ts

```diff
@@ -190,6 +190,23 @@ export const SignatureOptionsStruct = object({
   ),
 });
 
+export const NameLookupOptionsStruct = object({
+  /**
+   * The CAIP-2 chain ID. Defaults to `eip155:1`.
+   */
+  chainId: defaulted(string(), 'eip155:1'),
+
+  /**
+   * Domain name to lookup.
+   */
+  domain: optional(string()),
+
+  /**
+   * Address to lookup.
+   */
+  address: optional(string()),
+});
```

Comment:
```suggestion
export const BaseNameLookupOptionsStruct = object({
  /**
   * The CAIP-2 chain ID. Defaults to `eip155:1`.
   */
  chainId: defaulted(string(), 'eip155:1'),
});

export const NameLookupOptionsStruct = union([
  assign(BaseNameLookupOptionsStruct, object({ address: string() })),
  assign(BaseNameLookupOptionsStruct, object({ domain: string() })),
]);
```

This is one way to do it

Reply:
Cool. Updated and added tests for it.